src/SprykerEco/Zed/Payone/Business/Payment/Hook/PostSaveHook.php

```diff
@@ -154,7 +154,7 @@ protected function checkApiLogRedirectAndError(QuoteTransfer $quoteTransfer, Che
      */
     protected function checkApiAuthorizationRedirectAndError(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponse): CheckoutResponseTransfer
     {
-        $paymentEntity = $this->payoneRepository->createPaymentPayoneQueryByOrderId($checkoutResponse->getSaveOrder()->getIdSalesOrder())->findOne();
+        $paymentEntity = $this->payoneRepository->createPaymentPayoneQueryByOrderId($checkoutResponse->getSaveOrderOrFail()->getIdSalesOrderOrFail())->findOne();
```

Comment:
Same question as before.
> I'm not sure why we want to throw an exception here? This will interrupt the app for no obvious reason. Can't we cast this to string? What is the reason this is orFail()?

Reply:
In this case, I think we should leave it as is, because `createPaymentPayoneQueryByOrderId()` with 0 (if we don't have getIdSalesOrder()) doesn't make sense.

Reply:
The property is deprecated by the way, can you please check if it is possible to use the replacement?
`@deprecated Use SaveOrderTransfer directly/decoupled from the checkout response` Also, while we are here, the $checkoutResponse var should be renamed to $checkoutResponseTransfer.

Reply:
$checkoutResponse was changed to $checkoutResponseTransfer, but for deprecated property, unfortunately, we can't use directly/decoupled this transfer, because this method used in `executeCheckoutPostSaveHook` and `executePostSaveHook` and we have interfaces for this methods and we can't ask separate SaveOrderTransfer. It's how I see.